frontend/ui/widget/touchmenu.lua

```diff
@@ -547,9 +548,24 @@ function TouchMenu:init()
         self.page_info_right_chev
     }
     -- group for device info
-    self.time_info = TextWidget:new{
+    self.time_info = Button:new{
         text = "",
         face = self.fface,
+        text_font_bold = false,
+        callback = function()
+            local now_t = os.date("*t")
+            local time = util.secondsToHour(os.time(), G_reader_settings:isTrue("twelve_hour_clock"))
+            local message_text = T(_("Current date and time are:\n%1-%2-%3 at %4"),
+                now_t.year, now_t.month, now_t.day, time)
```

Comment:
I would drop the awkward wording and just display the date in the preferred format (e.g., `date +%c`). If we actually already had strftime in posix_h I'd have said use that, but given the edge-case, you cna probably just just an os.execute for `date` here (unless Android? In whihc case, strftime, for sure).

Reply:
Translation might be an issue, though, because most devices don't ship locales db, so the day/month names might be an issue (but I would *really* like to see them).

Reply:
Yeah, I was going to say, won't that pretty much only work on desktop…

Reply:
What about just using something like
```
local wday = os.date("%a")
local month = os.date("%b")
local day = os.date("%d")
local year = os.date("%Y")
-- ... and so on for all fields

day_names = {  -- for our translators
   Mon =  _("Mon"),
   Tue = _("Tue"),
}

local date_string = T_("%1 %2 %3 %4", day_name[wday], month_name[mon], day, year)
-- this format string has to be adjusted by translators
-- for german the format string would be "%1 %3 %2 %4"
local time_string
if use12hour_clock then
  time_string = T_("%1:%2:%3 PM"), hour, min, sec)
  -- to be adjusted
else
  time_String = T_("%1:%2:3"), hour, min, sec)
  -- to be adjusted
end

local message_text = T(_("%1 %2"), date_string, time_string) 
-- can be swapped by translators
```
and add some help text for translators to transform that string to their locale? The translators should know, what is usual. 
With this solution we can use the 12HourClock setting that is selected in KOReader.

Reply:
Yeah, that would be the overengineered approach, which unfortunately looks like the sanest approach in our context :/.

Reply:
If overengineering is needed, please let me try that :)

Reply:
There are a few week & month days already translated in the Statistics plugin (and its calendarview buddy).
(Though about suggesting that if you want to show the day date, showing a calendar with the month context is a better way to show the date - but lauching a fullscreen CalendarView from a tap on the clock on any menu feels really too funky. So, a simple text date in a popup is just fine.)